src/runtime/pickle_runner.ts

```diff
@@ -366,6 +380,24 @@ export default class PickleRunner {
     return await this.invokeStep(null, hookDefinition, hookParameter)
   }
 
+  async runStepHook(
+    stepHookDefinition: TestStepHookDefinition
+  ): Promise<messages.TestStepFinished.ITestStepResult> {
+    if (this.isSkippingSteps()) {
+      return messages.TestStepFinished.TestStepResult.fromObject({
+        status: Status.SKIPPED,
+      })
+    }
+    const hookParameter: ITestCaseHookParameter = {
+      gherkinDocument: this.gherkinDocument,
+      pickle: this.pickle,
+      testCaseStartedId: this.currentTestCaseStartedId,
+      result: this.result,
```

Comment:
I think this should have the cumulatedTestStepResult not the result of the pickle. 

Thoughts on also having the testStepId?

Reply:
Not sure exactly why we would want the cumulated? The beforeStep would always get an empty result (since nothing has run yet), and the afterStep Hook would want the pickle from the actual step definition, if I'm understanding right. I did try using cumulatedResult / updating the result, but wasn't successful in my attempts. 

I did go ahead and add the stestStepId.

Reply:
Thanks for adding testStepId. 

Agree for beforeStep it won't have a result.

For afterStep I think it makes more sense to have the stepResult for the following case:
* scenario with 2 steps, step 1 fails
* if afterStep gets scenario result, both step1 and step2 afterStep get failed result
* if afterStep gets step result, step1 afterStep gets failed result, step2 afterStep gets skipped result

I think the latter is more correct as the afterStep should only care about the current step. If you care about the scenario result, you can use the after hook.

Reply:
Figured out how to get this to work. Changed it so it sets stepResult to undefined at the start of each step (before running beforeStepHook so it doesn't maintain the stepResult from previous step.